src/main/java/org/jenkinsci/plugins/workflow/steps/TimeoutStepExecution.java

```diff
@@ -51,10 +53,18 @@ public boolean start() throws Exception {
 
     @Override
     public void onResume() {
-        super.onResume();
         setupTimer(System.currentTimeMillis());
     }
 
+    private TaskListener listener() {
+        try {
+            return getContext().get(TaskListener.class);
+        } catch (Exception x) {
+            LOGGER.log(Level.WARNING, null, x);
+            return TaskListener.NULL;
```

Comment:
When using injection, there were two ways `StepContext.get` errors were handled: `AbstractStepImpl.start` would throw up the exception, which we could still do in a `StepExecution` constructor; and `AbstractStepExecutionImpl.onResume` would send the exception to `StepContext.onFailure` (but presumably leave the field null, without preventing the rest of an `onResume` override from running—perhaps the root cause of JENKINS-34021 and the still-open [JENKINS-37486](https://issues.jenkins-ci.org/browse/JENKINS-37486)).

This idiom is more verbose but clearer—the step gets to decide what to do.

Alternately, we could change `onResume` to throw `Exception`. (Would still be binary-compatible.) `FlowExecutionList.ItemListenerImpl.onLoaded` would then need to catch exceptions and call `StepContext.onFailure`. Like `AbstractStepExecutionImpl`, this would rely on `transient` fields holding the contextual object; the improvement would be that a failure in `StepContext.get` would stop anything _else_ in `onResume` (such as `setupTimer` here) from running.